inspirehep/modules/workflows/views/callback.py

```diff
@@ -153,12 +179,79 @@ def webcoll_callback():
         )
         db.session.delete(pending_record)
         db.session.commit()
-        if response[recid].get('status') == 'failed':
-            continue
 
     return jsonify(response)
 
 
+def _robotupload_has_error(result):
+    recid = int(result.get('recid'))
+    if not result.get('success'):
+        message = result.get(
+            'error_message',
+            'No error message from robotupload.'
+        )
+    elif recid < 0:
+        message = result.get(
+            'error_message',
+            'Failed to create record on robotupload.',
+        )
+    else:
+        return False, ''
+
+    return True, message
+
+
+def _parse_robotupload_result(result, workflow_id):
+    response = {}
+    recid = int(result.get('recid'))
+
+    already_pending_ones = WorkflowsPendingRecord.query.filter_by(
+        record_id=recid,
+    ).all()
+    if already_pending_ones:
+        current_app.logger.warning(
+            'The record %s was already found on the pending list.',
+            recid
+        )
+        response = {
+            'status': 'failed',
+            'message': 'Recid %s already in pending list.' % recid,
+        }
+        return response
```

Comment:
Is it a critical error to the point of making the workflow to fail? Isn't this triggered whenever legacy sends twice the callback? Can't we simply ignore this?